components/load_from_hf_hub/fondant_component.yaml

```diff
@@ -19,11 +19,11 @@ args:
     description: Optional argument, a list containing the original image column names in case the 
       dataset on the hub contains them. Used to format the image from HF hub format to a byte string.
     type: list
-    default: []
+    default: None
```

Comment:
I am not sure I agree, then by definition it's either:
- a normal argument with [] as default and no longer optional
- Optional argument with [] as default which is not really common since `Optional` types don't really mix well with mutable data types
![image](https://github.com/ml6team/fondant/assets/47530815/acc1cc0b-4166-4d24-b05a-8a51f224299b)

Reply:
> * a normal argument with [] as default and no longer optional

If a default is defined, it is optional. It's just not KFP's `isOptional`. I don't see any issue with this.

> * Optional argument with [] as default which is not really common since `Optional` types don't really mix well with mutable data types
>   ![image](https://user-images.githubusercontent.com/47530815/274514827-acc1cc0b-4166-4d24-b05a-8a51f224299b.png)

In Python this is an issue indeed, however we don't need to define the default in Python. We can just define the default in the `fondant_component.yaml`:

**fondant_component.yaml**
```yaml
    default: []
```

**main.py**
```python
    image_column_names: list,
```

Reply:
We're just trying to indicate the absence of a value which is what `None` is better suited for. The empty list in this case won't be used for appending or modifying a certain behavior. Is there any added advantage compared to `None`? 

It also seems like we're making an arbitrary choice on which data types to define as having empty values. Should we also include dictionaries?

Reply:
The advantage is that you don't need to handle the `None` case in the code, and can always assume that the value is the of the type defined in the argument.

I would indeed include dictionaries as well. I don't think that's arbitrary.

This is btw just component implementation. Fondant supports `None` for `list` and `dict` as well. I just don't think we need to use it :stuck_out_tongue: